src/cmd/linuxkit/run_hyperkit.go

```diff
@@ -55,7 +64,48 @@ func runHyperKit(args []string) {
 		flags.Usage()
 		os.Exit(1)
 	}
-	prefix := remArgs[0]
+	path := remArgs[0]
+	prefix := path
+
+	_, err := os.Stat(path)
+	stat := err == nil
+
+	var isoPaths []string
+
+	// if the path does not exist, must be trying to do a kernel boot
+	if !stat {
+		_, err = os.Stat(path + "-kernel")
+		statKernel := err == nil
+		if statKernel {
+			*kernelBoot = true
+		} else {
+			log.Fatalf("Cannot find kernel file (%s): %v", path+"-kernel", err)
+		}
+		_, err = os.Stat(path + "-initrd.img")
+		statInitrd := err == nil
+		if !statInitrd {
+			log.Fatalf("Cannot find initrd file (%s): %v", path+"-initrd.img", err)
+		}
+	} else {
+		// if path ends in .iso they meant an ISO
+		if strings.HasSuffix(path, ".iso") {
+			*isoBoot = true
+			prefix = strings.TrimSuffix(path, ".iso")
+			// hyperkit only supports UEFI ISO boot at present
+			*uefiBoot = true
```

Comment:
you don't generally have to set this on qemu, as it will set it if the path has `-efi.iso` in, although I am wondering about making all the ISOs hybrid which would change this... The most consistent would probably be to set it if the path has `-efi.iso` in for now.